packages/core/test/app/common/utils/merkle-interval-tree.spec.ts

```diff
@@ -0,0 +1,274 @@
+import { should } from '../../../setup'
+
+/* External Imports */
+import BigNumber = require('bn.js')
+
+/* Internal Imports */
+import { MerkleIntervalTree } from '../../../../src/app/common/utils/merkle-interval-tree'
+
+import {
+  MerkleIntervalTreeLeafNode,
+  MerkleIntervalTreeInternalNode,
+  MerkleIntervalTreeInclusionProof,
+} from '../../../../src/interfaces/common/utils/merkle-interval-tree.interface'
+
+/**
+ * Converts a string to a hex buffer.
+ * @param value String value to convert.
+ * @returns the string as a hex buffer.
+ */
+const hexify = (value: string): Buffer => {
+  return Buffer.from(value, 'hex')
+}
+
+describe('MerkleIntervalTree', () => {
+  describe('construction', () => {
+    it('should be created correctly with no leaves', () => {
+      const tree = new MerkleIntervalTree()
+
+      const root = tree.getRoot()
+      const levels = tree.getLevels()
+
+      should.not.exist(root)
+      levels.should.deep.equal([[]])
+    })
+
+    it('should be created correctly with one leaf', () => {
+      const tree = new MerkleIntervalTree([
+        {
+          start: new BigNumber(0),
+          end: new BigNumber(100),
+          data: hexify('1234'),
+        },
+      ])
+
+      const root = tree.getRoot()
+
+      root.should.deep.equal({
+        index: new BigNumber(0),
+        hash: hexify(
+          '4d7654430bd809384e15ef3e842aad2449b6310b25c652a220af74716b37e0ae'
+        ),
+      })
+    })
+
+    it('should be created correctly with two leaves', () => {
+      const tree = new MerkleIntervalTree([
+        {
+          start: new BigNumber(0),
+          end: new BigNumber(100),
+          data: hexify('1234'),
+        },
+        {
+          start: new BigNumber(100),
+          end: new BigNumber(200),
+          data: hexify('5678'),
+        },
+      ])
+
+      const root = tree.getRoot()
+
+      root.should.deep.equal({
+        index: new BigNumber(0),
+        hash: hexify(
+          'e1b53cab461af771ad8d060145d2e27a04ee7c2e671efe4feac748de8cef1fc5'
+        ),
+      })
+    })
+  })
```

Comment:
Another test could check that an error is thrown if leaf nodes have invalid ranges (`start > end`)